peachjam/adapters/adapters.py

```diff
@@ -158,10 +179,78 @@ def update_document(self, url):
             # for stub documents, use the publication document as the source file
             pubdoc = document["publication_document"]
             if pubdoc and pubdoc["url"]:
-                self.download_source_file(pubdoc["url"], doc, title)
+                self.download_source_file(pubdoc["url"], created_doc, title)
         else:
             # the source file is the PDF version
-            self.download_source_file(f"{url}.pdf", doc, title)
+            self.download_source_file(f"{url}.pdf", created_doc, title)
+
+        self.fetch_relationships(document, created_doc)
+        logger.info("Fetching of relationships is complete!")
+
+    def fetch_relationships(self, imported_document, created_document):
+        subject_work = created_document.work
+
+        if imported_document["repeal"]:
+            repealing_work, _ = Work.objects.get_or_create(
+                frbr_uri=FrbrUri.parse(imported_document["repeal"]["repealing_uri"]),
```

Comment:
The database considers `frbr_uri` to be a string, so you don't want to parse it.

Reply:
Done